### PR TITLE
feat(tests): Setup Intern for functional tests

### PR DESCRIPTION
**Pull request by nchapman**

### CODE REVIEW

File: grunttasks/default.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+module.exports = function (grunt) {
+  'use strict';
+
+  grunt.registerTask('default', ['build', 'build']);
```
Comment (pdehaan) [inline grunttasks/default.js]: Explain this please...

Is the second "build" target just to be double sure?

Comment (nchapman) [inline grunttasks/default.js]: Haha oops!

File: tests/intern.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ // Learn more about configuring this file at <https://github.com/theintern/intern/wiki/Configuring-Intern>.
+// These default settings work OK for most people. The options that *must* be changed below are the
+// packages, suites, excludeInstrumentation, and (if you want functional tests) functionalSuites.
+define(['intern/lib/args'], function (args) {
+  'use strict';
+
+  return {
+    // Chronicle configuration
+    chronicle: {
+      url: 'http://localhost:8080/'
```
Comment (pdehaan) [inline tests/intern.js]: This is nice, but should we be pulling the domain/port via the convict config?

Comment (nchapman) [inline tests/intern.js]: Yeah it should. It doesn't like trying to do a regular old node `require` in this context, so I need to figure that out.

File: Gruntfile.js [hunk 3]
```diff
@@ -13,5 +13,6 @@ module.exports = function (grunt) {
     staticPath: config.get('server.staticPath')
   });
 
+  grunt.loadNpmTasks('intern');
```
Comment (pdehaan) [inline Gruntfile.js]: nit: should this be closer to the `require('load--grunt-tasks')(grunt);`?

File: grunttasks/default.js (hunk 1 above)
Comment (pdehaan) [inline grunttasks/default.js]: I JUST INCREASED PERFORMANCE BY 200%!!!

**Comment by nchapman** [comment]
This is on hold until we get the database sorted (#102).

File: tests/functional/visits.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  'intern/chai!expect'
+], function (intern, bdd, expect) {
+  'use strict';
+
+  var URL = intern.config.chronicle.url;
+
+  bdd.describe('visits', function () {
+    bdd.before(function () {
+      // login automatically as the fake user
+      this.remote.get(URL + '/auth/login');
```
Comment (6a68) [inline tests/functional/visits.js]: does this mean that intern automatically hangs on to the session cookies?

Comment (nchapman) [inline tests/functional/visits.js]: Yeah it's running selenium underneath so it's a regular browsing session.

Comment (6a68) [inline tests/functional/visits.js]: :open_mouth:

**Comment by 6a68** [comment]
I tried to run `npm install && npm test` locally and got a completely insane traceback. Known thing?

```
Running "intern:all" (intern) task
FATAL ERROR
TypeError: Object #<Object> has no method 'relative'
    at getSource (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/lib/util.js:372:22)
    at formatLine (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/lib/util.js:411:40)
    at processChromeTrace (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/lib/util.js:423:16)
    at normalizeStackTrace (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/lib/util.js:474:38)
    at Object.getErrorMessage (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/lib/util.js:582:14)
    at Object.logError (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/lib/util.js:625:23)
    at Object.consoleReporter./error (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/lib/reporters/console.js:33:9)
    at Object.target.(anonymous function).dispatcher [as on/error] (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/node_modules/dojo/aspect.js:91:38)
    at Function.on.emit (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/node_modules/dojo/on.js:281:37)
    at Object.Evented.emit (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/node_modules/dojo/Evented.js:31:19)
    ----------------------------------------
    rejected at signalDeferred (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/node_modules/dojo/Deferred.js:84:15)
    at signalListener (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/node_modules/dojo/Deferred.js:55:5)
    at signalWaiting (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/node_modules/dojo/Deferred.js:28:4)
    at reject.reject (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/node_modules/dojo/Deferred.js:219:5)
    at signalDeferred (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/node_modules/dojo/Deferred.js:84:15)
    at signalListener (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/node_modules/dojo/Deferred.js:58:4)
    at signalWaiting (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/node_modules/dojo/Deferred.js:28:4)
    at reject.reject (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/node_modules/dojo/Deferred.js:219:5)
    at /Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/node_modules/leadfoot/node_modules/dojo/Promise.js:235:37
    at runCallbacks (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/node_modules/leadfoot/node_modules/dojo/Promise.js:11:18)
    ----------------------------------------
[object Deferred]
    at Promise.then.promise.then (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/node_modules/dojo/Deferred.js:252:24)
    at Suite.run (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/lib/Suite.js:312:7)
    at /Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/main.js:54:18
    at /Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/lib/util.js:511:19
    at Array.forEach (native)
    at Object.run (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/main.js:53:16)
    at /Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/runner.js:280:12
    at execModule (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/node_modules/dojo/dojo.js:515:54)
    at /Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/node_modules/dojo/dojo.js:582:7
    at guardCheckComplete (/Users/jhirsch/codez/github/mozilla-chronicle/node_modules/intern/node_modules/dojo/dojo.js:566:4)
```

**Comment by nchapman** [comment]
> Known thing?

Not that I've seen :disappointed:.

**Comment by nchapman** [comment]
All right, I think this one is ready to go. I bailed on making the functional tests the default for `npm test` because of the dependencies. We can revisit this later. For now, Travis is running `grunt test`. We're using the new container system on Travis which seems to work really well and supports caching directories which definitely speeds up our builds, but maybe not as much as you might think. If you examine the output on Travis, you'll notice that creating a archive of our cached directories takes about 23 seconds. Overall though build times are good though, so I'm happy.

@6a68 @pdehaan will you give this another look when you have a moment?

**Pull request merged**